server/routes/getDashboards.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import {
+  IRouter,
+  IKibanaResponse,
+  ResponseError,
+} from '../../../../src/core/server';
+import { API_PREFIX } from '../../common';
+
+const XMLHttpRequest = require('xmlhttprequest').XMLHttpRequest;
+
+export default function (router: IRouter) {
+  router.get(
+    {
+      path: `${API_PREFIX}/getDashboards`,
+      validate: {},
+    },
```

Comment:
`get Dashboards` shouldn't be a handler implemented on the server side. Because 
1. no complex logic is added, just parsing json. 
2. need 2 http calls to achieve, one kibana api, one ES api.

Instead, do it in the client side, and only use one http call to ES.

Reply:
We cannot make a call to another server from the client-side, the call will have to be made from the server side. 
https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors

Reply:
There is a kibana rest api for retrieving saved_objects. Visualization/dashboards/saved_search are all saved objects. 
http://localhost:5601/api/saved_objects/_find?default_search_operator=AND&page=1&per_page=1000&search_fields=title%5E3&search_fields=description&type=dashboard

Reply:
frontend should communicate to elasticsearch with appropriate  security context.

https://www.elastic.co/guide/en/kibana/current/development-elasticsearch.html